src/raftstore/store/fsm/apply.rs

```diff
@@ -1768,14 +1766,16 @@ impl ApplyDelegate {
     // The target peer should send missing log entries to the source peer.
     //
     // So, the merge process order would be:
-    // 1. `exec_commit_merge` in target apply worker
-    // 2. `catch_up_logs_for_merge` in source apply worker (check whether need to catch up logs)
-    // 3. `on_ready_catch_up_logs` in source raftstore
-    // 4. ... (raft append and apply logs)
-    // 5. `on_ready_prepare_merge` in source raftstore (means source region has finished applying all logs)
-    // 6. `catch_up_logs_for_merge` in source apply worker (destroy itself and send LogsUpToDate)
-    // 7. resume `exec_commit_merge` in target apply worker
-    // 8. `on_ready_commit_merge` in target raftstore
+    // 1.   `exec_commit_merge` in target apply fsm
+    // 2.   `on_catch_up_logs_for_merge` in source peer fsm
+    // 3.   if the source peer has already executed the corresponding `on_ready_prepare_merge`, set pending_remove and jump to step 6
+    // 4.   ... (raft append and apply logs)
+    // 5.   `on_ready_prepare_merge` in source peer fsm and set pending_remove (means source region has finished applying all logs)
+    // 6.   `logs_up_to_date_for_merge` in source apply fsm (destroy its apply fsm and send Noop to trigger the target apply fsm)
+    // 7.   resume `exec_commit_merge` in target apply fsm
+    // 8.   `on_ready_commit_merge` in target peer fsm
+    // 9.   `on_merge_result` in source peer fsm (destroy its peer fsm and send Noop to trigger the target peer fsm)
```

Comment:
already trigger `on_ready_commit_mege` in step 8, why send `Noop` again?

Reply:
seems the first time calling of `on_ready_commit_merge` is to send `MergeResult`, and do actual work for the second calling.

Reply:
seems the first time calling of `on_ready_commit_merge` is to send `MergeResult`, and do actual work for the second calling?